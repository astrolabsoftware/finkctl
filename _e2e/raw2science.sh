#!/bin/bash

set -euxo pipefail

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"

export FINKCONFIG="$DIR"

mkdir -p $HOME/.kube
cat >> $HOME/.kube/config << EOF
apiVersion: v1
clusters:
- cluster:
    server: https://127.0.0.1:34729
  name: kind-kind
contexts:
- context:
    cluster: kind-kind
    user: kind-kind
  name: kind-kind
current-context: kind-kind
kind: Config
preferences: {}
users:
- name: kind-kind
  user:
EOF

finkctl run raw2science --image=param_image --dry-run > /tmp/raw2science.out

diff /tmp/raw2science.out $DIR/raw2science.out.expected
