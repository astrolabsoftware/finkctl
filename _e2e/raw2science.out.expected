spark-submit --master "k8s://https://127.0.0.1:34729" \
    --deploy-mode cluster \
    --conf spark.executor.instances=1 \
    --conf spark.kubernetes.namespace=default \
    --conf spark.kubernetes.authenticate.driver.serviceAccountName=spark \
    --conf spark.kubernetes.container.image="param_image" \
    --conf spark.driver.extraJavaOptions="-Divy.cache.dir=/tmp -Divy.home=/tmp" \
    --conf spark.hadoop.fs.s3a.endpoint=http://minio.minio:9000 \
    --conf spark.hadoop.fs.s3a.access.key="minioadmin" \
    --conf spark.hadoop.fs.s3a.secret.key="minioadmin" \
    --conf spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version=2 \
    --conf spark.hadoop.fs.s3a.connection.ssl.enabled=false \
    --conf spark.hadoop.fs.s3a.fast.upload=true \
    --conf spark.hadoop.fs.s3a.path.style.access=true \
    --conf spark.hadoop.fs.s3a.aws.credentials.provider=org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider \
    --conf spark.hadoop.fs.s3a.impl="org.apache.hadoop.fs.s3a.S3AFileSystem" \
    --conf spark.driver.memory=3Gi \
    --conf spark.executor.memory=3Gi \
    --conf spark.kubernetes.driver.request.cores=1 \
    --conf spark.kubernetes.executor.request.cores=1 \
    local:///home/fink/fink-broker/bin/raw2science.py \
    -log_level "INFO" \
    -online_data_prefix "s3a://fink-broker-online" \
    -producer "sims" \
    -tinterval "2" \
    -night "20200101"
